### PR TITLE
[GP-209] cache ANS name for owner in account page

### PR DESCRIPTION
**Pull request by 0xaptosj**
**Background**
Currently when we load the account page, we try to get ANS name for `sender` and `sent to` address of every tx.

e.g. for `account/0x123`, ANS `123.apt`, all txs will either have `sender` or `sent to` set to `0x123`, we end up converting `0x123` to `123.apt` many times.

**How does this PR work**
Add account owner's ANS name to `localStorage` with 5 min expiration.

### CODE REVIEW

**Comment by 0xaptosj** [comment]
we do not cache the other party of tx (e.g. sender is account owner, sent to is some other address) to avoid blowing up the localStorage

File: src/api/hooks/useGetANS.ts [hunk 1]
```diff
@@ -22,6 +22,12 @@ export function useGetNameFromAddress(address: string) {
   const [name, setName] = useState<string | undefined>();
 
   useEffect(() => {
+    const cachedName = localStorage.getItem(address)
```
Comment (jillxuu) [inline src/api/hooks/useGetANS.ts:25]: awesome lgtm! could you run `yarn fmt -w` to make sure format looks good?

Comment (jillxuu) [reply src/api/hooks/useGetANS.ts]: husky looks pretty solid to me. we will rewrite but mostly its gonna be to migrate to the nextjs framework. is husky compatible with pnpm thou? only npm & yarn are mentioned on the site.

Comment (0xaptosj) [reply src/api/hooks/useGetANS.ts]: looks husky [supports](https://typicode.github.io/husky/#/?id=automatic-recommended) pnpm, we can always try at least, but if migrating to nextjs will result in a new repo, then ill hold.

Comment (jillxuu) [reply src/api/hooks/useGetANS.ts]: nextjs migration will be done by third party so not sure at this point. lets try husky then

File: src/components/TitleHashButton.tsx [hunk 2]
```diff
@@ -114,6 +116,7 @@ function Name({address}: {address: string}) {
   if (!name) {
     return null;
   }
+  setLocalStorageWithExpiry(address, name, TTL);
```
[the remark this answers is not in the document]
Comment (0xaptosj) [reply src/components/TitleHashButton.tsx:119]: tested eviction by console log it.

**Review by jillxuu — APPROVED** [review]

File: src/components/TitleHashButton.tsx [hunk 3]
```diff
@@ -13,9 +13,11 @@ import ContentCopyIcon from "@mui/icons-material/ContentCopy";
 import useMediaQuery from "@mui/material/useMediaQuery";
 import {truncateAddress, truncateAddressMiddle} from "../pages/utils";
 import {useGetNameFromAddress} from "../api/hooks/useGetANS";
+import {setLocalStorageWithExpiry} from "../utils";
 
 const BUTTON_HEIGHT = 34;
 const TOOLTIP_TIME = 2000; // 2s
+const TTL = 300000; // 5 minutes
```
[the remark this answers is not in the document]
Comment (0xaptosj) [reply src/components/TitleHashButton.tsx]: updated to 1 minute to account for user changes their ANS name.

**Pull request merged**